### PR TITLE
[fix] Display subcategories in blog layout only when user has access

### PR DESCRIPTION
**Pull request by infograf768**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/18397

### Summary of Changes
Check access to subcategories.

### Testing Instructions
See https://github.com/joomla/joomla-cms/issues/18397

Before patch, subcategories display even if they are set to registered access. Clicking on their title gets a 403.

After patch the registered subcategories do not display until the user has logged in.

### CODE REVIEW

**Comment by infograf768** [comment]
Indentation had to be changed which explains why this patch looks so big.
real code changes are very small.

**Comment by brianteeman** [comment]
@infograf768 there is only one changed line in the pr

**Comment by infograf768** [comment]
Nope. You have only seen `+$user  = JFactory::getUser();`
look further and please test
```
+		<?php $groups = $user->getAuthorisedViewLevels(); ?>
+		<?php // Check whether category access level allows access to subcategories. ?>
+		<?php if (in_array($child->access, $groups)) : ?>
```
and obvioulsy a new endif before endforeach

File: components/com_content/views/category/tmpl/blog_children.php [hunk 1]
```diff
@@ -13,69 +13,73 @@
...
-			<?php if ($lang->isRtl()) : ?>
-			<h3 class="page-header item-title">
-				<?php if ( $this->params->get('show_cat_num_articles', 1)) : ?>
-					<span class="badge badge-info tip hasTooltip" title="<?php echo JHtml::_('tooltipText', 'COM_CONTENT_NUM_ITEMS_TIP'); ?>">
-						<?php echo $child->getNumItems(true); ?>
-					</span>
-				<?php endif; ?>
-				<a href="<?php echo JRoute::_(ContentHelperRoute::getCategoryRoute($child->id)); ?>">
-				<?php echo $this->escape($child->title); ?></a>
+		<?php $groups = $user->getAuthorisedViewLevels(); ?>
```
Comment (csthomas) [inline components/com_content/views/category/tmpl/blog_children.php]: It would be better to put it higher up to line 17 or 19

Comment (infograf768) [reply components/com_content/views/category/tmpl/blog_children.php]: Done.

**Comment by infograf768** [comment]
Needs tests, folks. thanks.

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 3e3467103c080d608c76098fc9a397e0c3b5ba1a<br /><br />tested successfully for article category blog but shouldnt the same change be made for article category list

luckily the contacts component does not have the same issue<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18408">issues.joomla.org/tracker/joomla-cms/18408</a>.</sub>

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 3e3467103c080d608c76098fc9a397e0c3b5ba1a<br /><br />It would be good to do the same for beez3 template?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18408">issues.joomla.org/tracker/joomla-cms/18408</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Comment by infograf768** [comment]
yes to both i.e list and beez.

**Comment by mbabker** [comment]
Based on comments does this need further updates?

**Comment by infograf768** [comment]
Will try to update today for list and beez, if OK for you @mbabker

**Comment by infograf768** [comment]
Updated:
Added similar code for category list and blog/list for Beez.

@brianteeman 
@csthomas 
Please test again

**Comment by infograf768** [comment]
Restarted travis as php 5.6 did not complete.

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 6a6e91d71d393250303529cee4a4ceffcf4e1b37<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18408">issues.joomla.org/tracker/joomla-cms/18408</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 6a6e91d71d393250303529cee4a4ceffcf4e1b37<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/18408">issues.joomla.org/tracker/joomla-cms/18408</a>.</sub>

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**